etcdserver/v3_server.go

```diff
@@ -86,26 +89,28 @@ type Authenticator interface {
 }
 
 func (s *EtcdServer) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResponse, error) {
-	if r.Serializable {
-		var resp *pb.RangeResponse
-		var err error
-		chk := func(ai *auth.AuthInfo) error {
-			return s.authStore.IsRangePermitted(ai, r.Key, r.RangeEnd)
+	if !r.Serializable {
+		notify, err := s.linearizableReadNotify(ctx)
+		if err != nil {
+			return nil, err
 		}
-		get := func() { resp, err = s.applyV3Base.Range(noTxn, r) }
-		if serr := s.doSerialize(ctx, chk, get); serr != nil {
-			return nil, serr
+
+		select {
+		case <-notify:
+		case <-ctx.Done():
+			return nil, ctx.Err()
 		}
-		return resp, err
 	}
-	result, err := s.processInternalRaftRequest(ctx, pb.InternalRaftRequest{Range: r})
-	if err != nil {
-		return nil, err
+	var resp *pb.RangeResponse
```

Comment:
nm, I see it falls through from linearized to serialized once it gets the notify